Makefile

```diff
@@ -109,7 +109,7 @@ wait:
 # More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
 docker-build: generate ## Build docker image with the operator-controller.
-	docker build -t ${IMG} .
+	docker build -t ${IMG} -f Dockerfile ./bin/
```

Comment:
This seems like it'll fail because the Dockerfile expects the binary to be pre-built by goreleaser. The previous Dockerfile included the binary build as a pre-step in a multi-stage build.

I think we need to add `build` to the list of pre-reqs?

Reply:
Also, the binary built in this case needs to be GOOS=linux, but in the non-docker build case should just take the default settings of the environment. That way, mac and linux machines can both build local binaries and container image binaries correctly.

Reply:
I've updated the `build` target to use `goreleaser build`, and I believe that's also resolved your concern here. By default, when running `make build`, goreleaser will set GOARCH and GOOS to the user's setup. This won't apply though when building for release.

Reply:
For `docker-build` I _think_ we need to explicitly set GOOS=linux, right? If we don't won't a macOS user end up with a docker container that contains a darwin binary?

Reply:
TBH, building within the container generally improves compatibility within the container and improves build reproducibility.

Reply:
GoReleaser doesn't support that, so if we want binary builds happening inside docker, we can't use goreleaser I don't think.

I don't love this restriction, but my main interests are:
- we have exactly one way to build the the binaries snd containers 
- the binaries we build ARE the binaries that end up in the container.

GoReleaser can be made to ensure reproducibility (outside of Go compiler version) so I think we end up at the same place. It just takes a bit more care to get there.